adage/analyze/management/commands/import_gene_sample_expr.py

```diff
@@ -0,0 +1,140 @@
+        except Exception as e:
+            raise CommandError(
+                "Raised exception when importing gene-sample expression "
+                "values: %s" % e)
+
+
+def import_expr(file_handle):
+    """
+    Function that reads input file and load gene-sample expression values
+    into the database.
+    """
+
+    # Enclose reading/importing process in a transaction context manager.
+    # Any exception raised inside the manager will terminate the transaction
+    # and roll back the database.
+    with transaction.atomic():
+        samples = []
+        for line_index, line in enumerate(file_handle):
+            tokens = line.rstrip('\r\n').split('\t')
+            if line_index == 0:
+                tokens = tokens[1:]
+                read_header(tokens, samples)
+            else:
+                import_data_line(line_index + 1, tokens, samples)
+
+
+def read_header(tokens, samples):
+    """
+    Read input tokens on header line and save the corresponding sample
+    object into "samples". (Each token will be searched in the database
+    using "ml_data_source" field. If a token does not match any sample's
+    ml_data_source, put None into "samples".)
+
+    An exception will be raised if any of the following errors is detected:
+      * Sample token is blank (null or consists of space characters only);
+      * Sample token is duplicate;
+    """
+
+    token_set = set()
+    for index, data_source in enumerate(tokens):
+        if not data_source or data_source.isspace():
+            raise Exception("Input file line #1 column #%d: blank data_source"
+                            % index + 2)
+        elif data_source in token_set:
+            raise Exception("Input file line #1 column #%d: %s is duplicate" %
+                            (index + 2, data_source))
+        else:
+            try:
+                token_set.add(data_source)
+                sample = Sample.objects.get(ml_data_source=data_source)
+                samples.append(sample)
+            except Sample.DoesNotExist:
+                samples.append(None)
+                logger.warn(
+                    "Input file line #1: data_source in column #%d not found "
+                    "in the database: %s", index + 2, data_source)
+
+
+def import_data_line(line_num, tokens, samples):
+    """
+    Function that imports numerical values in input tokens into the database.
+    An exception will be raised if any of the following errors is detected:
+      * The number of columns on this line is not equal to the number of
+        samples plus 1.
+      * The gene's "systematic_name" field (column #1) is blank;
+      * Data field (from column #2 to the end) can not be converted into a
+        float type.
+    """
+
+    if len(tokens) != len(samples) + 1:
+        raise Exception("Input file line #%d: Number of columns is not %d" %
+                        (line_num, len(samples) + 1))
```

Comment:
Thank you for the recommendation of `The Elements of Style` book, it is definitely something good to know. But I am not 100% convinced that the rule in Section I, part 9 applies to the current scenario, because in "any of the following errors is/are detected", the subject is "any", which could refer to either singular or plural, while in:
```One of the ablest men who have attacked this problem```
"who" clearly refers to "men", which is plural. 
As you said, this is just a comment line and not that important, so I will change it into plural.